server/src/main/java/io/druid/client/cache/LocalCacheProvider.java

```diff
@@ -33,13 +40,60 @@
   @Min(0)
   private int initialSize = 500000;
 
-  @JsonProperty
-  @Min(0)
-  private int logEvictionCount = 0;
+  @JacksonInject
+  private DruidProcessingConfig config = null;
 
   @Override
   public Cache get()
   {
-    return new MapCache(new ByteCountingLRUMap(initialSize, logEvictionCount, sizeInBytes));
+    if(sizeInBytes > 0) {
+      return MapCache.create(sizeInBytes, initialSize, config.getNumThreads());
```

Comment:
If you set the concurrency to the number of threads, the likelihood of having a collision in the striped locking is pretty high if they happen to each request a random lock. Can we make the concurrency higher than the number of threads?